project_recalculate/models/resource_calendar.py

```diff
@@ -25,13 +26,49 @@ def get_working_days_of_date(self, start_dt=None, end_dt=None,
                 end_day = current.replace(hour=23, minute=59, second=59)
                 end = end_dt if end_day > end_dt else end_day
                 obj = self.with_context(tz='UTC')
-                working_intervals = obj.get_working_intervals_of_day(
-                    start_dt=current, end_dt=end, leaves=leaves,
-                    compute_leaves=compute_leaves, resource_id=resource_id,
-                    default_interval=default_interval,
-                )
-                if working_intervals:
+                working_intervals = obj._work_intervals(current, end, resource)
+                if working_intervals._items:
                     days += 1
             next_dt = current + timedelta(days=1)
             current = next_dt
         return days
+
+    @api.multi
+    def plan_days_to_resource(self, days, day_dt, compute_leaves=False,
+                              resource=None, domain=None):
+        day_dt, revert = make_aware(day_dt)
+
+        # which method to use for retrieving intervals
+        if compute_leaves:
+            get_intervals = partial(self._work_intervals, resource=resource,
```

Comment:
Why accessing protected variable `_work_intervals`?

Reply:
I have added the `plan_days_to_resource` method to resource.calendar with the intention that it does the same as plan_days but for a specific resource that is passed by parameter. Therefore, the code of that method is almost the same as that of the existing plan_days method, that's why there is the call to the protected method _work_intervals.
I have done this with the intention that the planning of the days take into account the leaves of the resource